### PR TITLE
Added Camel Case support for labels and properties.

### PR DESCRIPTION
**Pull request by SyntaxUnknown**
Included the ability to provide camel case support for labels and properties. When creating a model, you can specify `.WithCamelcase()` to enable this support.

### CODE REVIEW

File: ExRam.Gremlinq.Core/Models/IGraphModel.cs [hunk 1]
```diff
@@ -8,5 +9,9 @@ public interface IGraphModel
 
         IGraphElementModel VerticesModel { get; }
         IGraphElementModel EdgesModel { get; }
+
+        object GetIdentifier(Expression expression);
+
+        object GetIdentifier(Type elementType, string memberName);
```
Comment (danielcweber) [inline ExRam.Gremlinq.Core/Models/IGraphModel.cs]: Looks very good. I'd rather not have two ways of getting an identifier here though, the former should do.

File: ExRam.Gremlinq.Core/Models/GraphModel.cs [hunk 2]
```diff
@@ -21,7 +22,103 @@ private enum GraphElementType
...
+            private sealed class CamelcaseGraphElementModel : IGraphElementModel
+            {
+                private readonly IGraphElementModel _baseModel;
+
+                public CamelcaseGraphElementModel(IGraphElementModel baseModel)
+                {
+                    _baseModel = baseModel;
+                }
+
+                public Option<string> TryGetConstructiveLabel(Type elementType) => _baseModel.TryGetConstructiveLabel(elementType).Map(x => x.ToCamelCase());
```
Comment (danielcweber) [inline ExRam.Gremlinq.Core/Models/GraphModel.cs]: I'd like to separate lowercase labels from lowercase identifiers, and so for CamelCase. So could we have a `WithCamelcaseIdentifiers` extension and a `WithCamelcaseLabels` extension, if you need both?

Comment (danielcweber) [reply ExRam.Gremlinq.Core/Models/GraphModel.cs]: or just the latter, if you just need labels in camel case.

File: ExRam.Gremlinq.Core/Models/IGraphModel.cs (hunk 1 above)
Comment (SyntaxUnknown) [reply ExRam.Gremlinq.Core/Models/IGraphModel.cs]: Ok, I removed the 2nd method and updated any references. Now, only `object GetIdentifier(Expression expression)` is exposed.

File: ExRam.Gremlinq.Core/Models/GraphModel.cs (hunk 2 above)
Comment (SyntaxUnknown) [reply ExRam.Gremlinq.Core/Models/GraphModel.cs]: I broke out `WithCamelcase` into `WithCamelcaseLabels` and `WithCamelcaseIdentifiers`. I also added in some tests to confirm it works in either order of adding them to the model.

**Comment by danielcweber** [comment]
Thanks a lot.

**Pull request merged**